src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,168 @@
+            case "todo":
+                matcher = this.regexParse("^todo\\s([\\w\\s]*)$", response);
+                if (!matcher.find() || matcher.groupCount() != 1) {
+                    throw new InvalidTaskException("Invalid use of todo. Usage: todo <task description>");
+                }
+                return new ToDoCommand(matcher.group(1), false);
+            case "deadline":
+                matcher = regexParse("^deadline\\s([\\w\\s]*)\\s\\/by\\s([\\w\\s\\:\\-]*)$", response);
+                if (!matcher.find() || matcher.groupCount() != 2) {
+                    throw new InvalidTaskException(
+                            "Invalid use of deadline. Usage: deadline <task description> /by <date & time>"
+                    );
+                }
+                return new DeadlineCommand(matcher.group(1), matcher.group(2), false);
+            case "event":
+                matcher = regexParse("^event\\s([\\w\\s]*)\\s\\/from\\s([\\w\\s\\-\\:]*)\\s\\/to\\s([\\w\\s\\-\\:]*)$", response);
+                if (!matcher.find() || matcher.groupCount() != 3) {
+                    throw new InvalidTaskException(
+                            "Invalid use of event. Usage: event <task description> /from <date & time> /to <date & time>"
+                    );
+                }
+                return new EventCommand(matcher.group(1), matcher.group(2), matcher.group(3), false);
+            case "delete":
+                matcher = regexParse("^delete\\s([\\w\\s]*)$", response);
+                if (!matcher.find() || matcher.groupCount() != 1) {
+                    throw new InvalidTaskException(
+                            "Invalid input. Usage: delete <task_index>"
+                    );
+                }
+                return new DeleteCommand(Integer.parseInt(matcher.group(1)));
+            case "save":
+                return new SaveCommand(response);
+            case "load":
+                return new LoadCommand(response);
+            case "find":
+                matcher = this.regexParse("^find\\s([\\w\\s]*)$", response);
+                if (!matcher.find() || matcher.groupCount() != 1) {
+                    throw new InvalidTaskException(
+                            "Invalid input. Usage: find <description to match>"
+                    );
+                }
+                return new FindCommand(matcher.group(1));
+            default: {
+                return new DefaultCommand(response);
+            }
+        }
+    }
+
+    /**
+     * Parses the lines of the save file and returns a Task to be added to the TaskList
+     * @param response line of the save file to be parsed
+     * @return A task generated from the saved line
+     * @throws InvalidFileTypeException
+     */
+    public Task parseSave(String response) throws InvalidFileTypeException {
+        boolean done;
+        String[] responseList = response.split("\\|");
+        Task newtask;
```

Comment:
Does the variable name newtask follow the java naming style of using the camel case syntax?